File: Skyscraper-iOS-20251106/Skyscraper/Skyscraper/Models/StarterPackModels.swift
//
//  StarterPackModels.swift
//  Skyscraper
//
//  Models for BlueSky Starter Packs
//

import Foundation

struct StarterPack: Codable, Identifiable {
    let uri: String
    let cid: String
    let record: StarterPackRecord
    let creator: Author
    let listItemCount: Int?
    let joinedWeekCount: Int?
    let joinedAllTimeCount: Int?
    let labels: [Label]?
    let indexedAt: String?

    var id: String { uri }
}

struct StarterPackRecord: Codable {
    let name: String
    let description: String?
    let descriptionFacets: [Facet]?
    let list: String
    let feeds: [FeedItem]?
    let createdAt: String
}

struct FeedItem: Codable, Identifiable {
    let uri: String

    var id: String { uri }
}

struct StarterPacksResponse: Codable {
    let starterPacks: [StarterPack]
    let cursor: String?
}

File: app/src/main/java/com/cameronbanga/skyscraper/models/StarterPackModels.kt
package com.cameronbanga.skyscraper.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * BlueSky Starter Pack model
 */
@JsonClass(generateAdapter = true)
data class StarterPack(
    val uri: String,
    val cid: String,
    val record: StarterPackRecord,
    val creator: Author,
    val listItemCount: Int? = null,
    val joinedWeekCount: Int? = null,
    val joinedAllTimeCount: Int? = null,
    val labels: List<Label>? = null,
    val indexedAt: String? = null
)

/**
 * StarterPack record data
 */
@JsonClass(generateAdapter = true)
data class StarterPackRecord(
    val name: String,
    val description: String? = null,
    val descriptionFacets: List<Facet>? = null,
    val list: String,
    val feeds: List<FeedItem>? = null,
    val createdAt: String
)

/**
 * Feed item reference
 */
@JsonClass(generateAdapter = true)
data class FeedItem(
    val uri: String
)

/**
 * Response containing list of starter packs
 */
@JsonClass(generateAdapter = true)
data class StarterPacksResponse(
    val starterPacks: List<StarterPack>,
    val cursor: String? = null
)

/**
 * List item view containing user information
 */
@JsonClass(generateAdapter = true)
data class ListItemView(
    val uri: String,
    val subject: Author
)

/**
 * Response for list items
 */
@JsonClass(generateAdapter = true)
data class ListItemsResponse(
    val cursor: String? = null,
    val items: List<ListItemView>
)

/**
 * List response (alias for ListItemsResponse)
 */
typealias ListResponse = ListItemsResponse
